src/linesearches_compat.jl

```diff
@@ -0,0 +1,234 @@
+    p_tmp = copy(M, p)
+    X_tmp = copy(M, p, η)
+    Y_tmp = copy(M, p, η)
+    f = get_cost_function(get_objective(mp))
+    dphi_0 = real(inner(M, p, X, η))
+    #println("dphi_0 = ", dphi_0)
+
+    # guess initial alpha
+    get_initial_alpha(M, css.initial_step, cs, p, η, fp, dphi_0)
+    α0 = cs.alpha
+
+    cs.last_f_p = fp
+
+    # perform actual linesearch
+
+    function ϕ(α)
+        retract!(M, p_tmp, p, η, α, css.retraction_method)
+        return f(M, p_tmp)
+    end
+    function dϕ(α)
+        retract!(M, p_tmp, p, η, α, css.retraction_method)
+        get_gradient!(mp, X_tmp, p_tmp)
+        vector_transport_to!(M, Y_tmp, p, η, p_tmp, css.vector_transport_method)
+        return real(inner(M, p_tmp, Y_tmp, Y_tmp))
+    end
+    function ϕdϕ(α)
+        # TODO: optimize?
+        retract!(M, p_tmp, p, η, α, css.retraction_method)
+        get_gradient!(mp, X_tmp, p_tmp)
+        vector_transport_to!(M, Y_tmp, p, η, p_tmp, css.vector_transport_method)
+        phi = f(M, p_tmp)
+        dphi = real(inner(M, p_tmp, Y_tmp, Y_tmp))
+        return (phi, dphi)
+    end
+
+    try
+        α, fp = css.linesearch(ϕ, dϕ, ϕdϕ, α0, fp, dphi_0)
+        return α
+    catch ex
+        if isa(ex, LineSearches.LineSearchException)
+            # maybe indicate failure?
+            return zero(dphi_0)
+        else
+            rethrow(ex)
+        end
+    end
+end
+
+function show(io::IO, cs::LineSearchesStepsize)
+    return print(
+        io,
+        "LineSearchesStepsize($(cs.linesearch); retraction_method=$(cs.retraction_method), vector_transport_method=$(cs.vector_transport_method))",
+    )
+end
+
+"""
+    nanmax(x, y)
+
+Maximum of arguments, ignoring those that are NaN.
+"""
+function nanmax(x, y)
+    isnan(x) && return y
+    isnan(y) && return x
+    return max(x, y)
+end
+function nanmax(x, y, z)
+    return nanmax(nanmax(x, y), z)
+end
+
+"""
+    nanmin(x, y)
+
+Minimum of arguments, ignoring those that are NaN.
+"""
+function nanmin(x, y)
+    isnan(x) && return y
+    isnan(y) && return x
+    return min(x, y)
+end
+
+# code below was adapted from https://github.com/JuliaNLSolvers/LineSearches.jl/blob/master/src/initialguess.jl
+
+function get_initial_alpha(
+    M::AbstractManifold, is::InitialStatic{T}, state::StepsizeStorage, p, η, phi_0, dphi_0
+) where {T}
+    PT = promote_type(T, real(number_eltype(η)))
+    if is.scaled == true && (ns = real(norm(M, p, η))) > convert(PT, 0)
+        # TODO: Type instability if there's a type mismatch between is.alpha and ns?
+        state.alpha = convert(PT, min(is.alpha, ns)) / ns
+    else
+        state.alpha = convert(PT, is.alpha)
+    end
+end
+
+function get_initial_alpha(
+    ::AbstractManifold, is::InitialPrevious, state::StepsizeStorage, p, η, phi_0, dphi_0
+)
+    if isnan(state.alpha)
+        state.alpha = is.alpha
+    end
+    state.alpha = min(is.alphamax, state.alpha)
+    return state.alpha = max(is.alphamin, state.alpha)
+end
+
+function get_initial_alpha(
```

Comment:
also here – no clue what this does, also there are comments in the code that read like todos or issues. I would prefer to not have such comments.